workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/utils/getMDFListColumns.tsx

```diff
@@ -83,6 +86,47 @@ export default function getMDFListColumns(
 					});
 				}
 
+				if (currentValue === PermissionActionType.CANCEL && row[MDFColumnKey.STATUS] === Status.APPROVED.name) {
+					const {data: mdfRequest} = useGet<
+						MDFRequestDTO
+					>(
+						row[MDFColumnKey.ID] &&
+							`/o/${LiferayAPIs.OBJECT}/${ResourceName.MDF_REQUEST_DXP}/${row[MDFColumnKey.ID]}?nestedFields=mdfReqToActs%2C%20mdfReqToMDFClms`
+					);
+					previousValue.push({
+						icon: 'block',
+						key: Status.CANCELED.key,
+						label: ' Cancel',
+						onClick: () => {
+							Liferay.Util.openConfirmModal({
+								message:
+									'Are you sure you want to cancel the MDF request?',
+								onConfirm: async (isConfirmed: boolean) => {
+									if (isConfirmed) {
+										const newRequestStatus = await patchRequestStatus(
+											Status.CANCELED,
+											String(row[MDFColumnKey.ID]),
+											mdfRequest?.mdfReqToActs,
+											mdfRequest?.mdfReqToMDFClms
+										);
```

Comment:
```suggestion
									if (isConfirmed) {
										const newRequestStatus = await patchRequestStatus(
											Status.CANCELED,
											String(row[MDFColumnKey.ID])
										);
```

Reply:
Couldn't refactor patchRequestStatus function (see print in message above).

Reply:
Done!